src/app/page.js

```diff
@@ -16,30 +26,34 @@ export default function Home() {
       .then((response) => response.json())
       .then((responseBody) => {
         const municipalDataObject = responseBody.data;
-        setMunicipalData(municipalDataObject);
+        setMunicipalData(municipalDataObject)
         setLoading(false);
       });
   }, []);
 
-  const index = municipalData.map((item) => {
-    item["municipality.long_name"];
-  });
   const renderName = municipalData.map((item) => (
-    <div key={item["municipality.name"]}>
+    <div key={item["municipality.name"]} name={item["municipality.province_name"]}>
       <a href={item["municipality.url"]}>
-        {item["municipality.name"] +
-          "," +
-          " " +
-          item["municipality.demarcation_code"]}
+        <div>{item["municipality.name"]}</div>
+        <div>{item["municipality.demarcation_code"]}</div>
       </a>
     </div>
   ));
+  
 
   return (
     <main>
       <div>
-        <MunicipalDataList renderName={renderName} />
         <div>{loading ? <>Loading..</> : <></>}</div>
+        <EasternCapeProvince municipalData={municipalData}  />
+        <FreestateProvince municipalData={municipalData}  />
+        <GautengProvince municipalData={municipalData}  />
+        <KwazuluNatalProvince municipalData={municipalData}  />
+        <LimpopoProvince municipalData={municipalData}  />
+        <MpumalangaProvince municipalData={municipalData}  />
+        <NorthernCapeProvince municipalData={municipalData}  />
+        <NorthWestProvince municipalData={municipalData}  />
+        <WesternCapeProvince municipalData={municipalData}  />
```

Comment:
Instead of passing the entire list of municipalities to a dedicated component per province, I would group the data by province first, then handle each province group at a time.

Something like

```javascript
provinceGroups = _.groupBy(municipalData.data, muni => muni["municipality.province_name"]);
```

for the grouping, and then you end up with something like

`{"Eastern Cape": [{municipality.demarcation_code": "BUF", ...}], ...}`

Then you can map over the province groups like

```jsx
{provinceGroups.keys().map(provinceName => {
    <ProvinceSection name={provinceName} municipalities={provinceGroups[provinceName]} />
}}
```

Reply:
Province section would then render the province name, and map over the municipalities in the group to render each of them

Reply:
https://lodash.com/docs/4.17.15#groupBy